packages/optimizer/lib/transformers/OptimizeAmpBind.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {nextNode, firstChildByTag, setAttribute} = require('../NodeUtils');
+const {skipNodeAndChildren} = require('../HtmlDomHelper');
+const {isTemplate} = require('../AmpConstants');
+
+/**
+ */
+class OptimizeAmpBind {
+  constructor(config) {
+    this.log_ = config.log.tag('OptimizeAmpBind');
+
+    // setup implementation only if placeholder generation is enabled
+    this.enabled_ = !!config.optimizeAmpBind;
```

Comment:
We should enable this by default.

```suggestion
    this.enabled_ = config.optimizeAmpBind !== false;
```

Reply:
Or did you intentionally leave this disabled until the validator changes are live?

Reply:
Yup, this is intentionally disabled until validator changes are live in AMP Cache. I was gonna follow up with a PR to enable it afterwards.